src/coreclr/jit/codegencommon.cpp

```diff
@@ -3341,6 +3344,26 @@ void CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg, bool* pXtraRegClobbere
                     assert(regArgTab[argNum].slot <= (int)varDsc->lvHfaSlots());
                     assert(argNum > 0);
                     assert(regArgTab[argNum - 1].varNum == varNum);
+
+                    // If the field is passed in the same register as the destination,
+                    // but is in the wrong part of the register, mark it specially so later
+                    // we make sure to move it to the right spot before "freeing" the destination.
+                    //
+                    // If the arg is already in the right place we don't need to do anything special
+                    destRegNum = varDsc->GetRegNum();
+                    if (regNum == destRegNum)
+                    {
+                        const int slot = regArgTab[argNum].slot;
+                        if (slot != 1)
```

Comment:
The condition above, `else if (regArgTab[argNum].slot == 1)` has already been checked. So won't this check always pass?

Reply:
Yes, looks like it will; let me change this to an assert.